staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -388,7 +388,7 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 
 	elems := make([]*elemForDecode, 0, len(getResp.Kvs))
 	for _, kv := range getResp.Kvs {
-		data, _, err := s.transformer.TransformFromStorage(kv.Value, authenticatedDataString(key))
+		data, _, err := s.transformer.TransformFromStorage(kv.Value, authenticatedDataString(kv.Key))
```

Comment:
The intention is to have the location of the resource as the key. Since the key was written using its own location, it must be authenticated (while decrypting) using its own location as well. `kv.Key` is its location. `key` was a top level location (example: `/api/Secrets/`). When I was attempting to write integration tests (which couldn't get done at that time though), I discovered this issue.

Let me know if I went wrong somewhere.